File: data/data.code.typescript.client/AckRoleRandNameOk.ts
namespace proto {
export class AckRoleRandNameOk
{
	private _uname: string;


	constructor(packet?: game.util.Packet) {
		if (packet) {
			this._uname = packet.ReadString();
		}
	}


	public get uname(): string { return this._uname; }
	public set uname(value: string) { this._uname = value; }
}
}

File: data/data.code.golang.client/ack_role_rand_name_ok.go
package proto

import (
	"packet"
)

type AckRoleRandNameOk struct {
	Uname                    string
}

func AckRoleRandNameOkDecode(pack *packet.Packet) *AckRoleRandNameOk {
	ackRoleRandNameOk := &AckRoleRandNameOk{}

	ackRoleRandNameOk.Uname = pack.ReadString()
	return ackRoleRandNameOk
}
